boa/src/value/operations.rs

```diff
@@ -394,7 +394,8 @@ impl JsValue {
                 Err(_) => f64::NAN,
             }),
             Self::Rational(num) => Self::new(-num),
-            Self::Integer(num) => Self::new(-f64::from(num)),
+            Self::Integer(num) if num == 0 => Self::new(-f64::from(0)),
```

Comment:
```suggestion
            Self::Integer(num) if num == 0 => Self::new(f64::from(0)),
```
This is returning `-0.0f64` when `num == 0`, which is incorrect; the negation of `0i32` is `0i32`. That's probably why the tests fail.